vector.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2025 The Godror Authors
+//
+//
+// SPDX-License-Identifier: UPL-1.0 OR Apache-2.0
+
+package godror
+
+/*
+#include <stdlib.h>
+#include "dpiImpl.h"
+
+// C Wrapper Function to Set the Dimensions Union Field
+void setVectorInfoDimensions(dpiVectorInfo *info, void *ptr) {
+    info->dimensions.asPtr = ptr;
+}
+
+// C Wrapper Function to Get the Dimensions Union Field
+void* getVectorInfoDimensions(dpiVectorInfo *info) {
+    return info->dimensions.asPtr;
+}
+
+*/
+import "C"
+
+import (
+	"fmt"
+	"unsafe"
+)
+
+// Vector holds the embedding VECTOR column starting from 23ai.
+type Vector struct {
+	Dimensions uint32      // Total dimensions of the vector.
+	Indices    []uint32    // Indices of non-zero values (sparse format).
+	Values     interface{} // Non-zero values (sparse format) or all values (dense format).
+	IsSparse   bool        // Flag to detect if it's a sparse vector
+}
+
+// SetVectorValue converts a Go `Vector` into a godror data type.
+func SetVectorValue(c *conn, v *Vector, data *C.dpiData) error {
+	var vectorInfo C.dpiVectorInfo
+	var valuesPtr unsafe.Pointer
+	var format C.uint8_t
+	var dimensionSize uint8
+	var numDims int
+
+	switch values := (*v).Values.(type) {
+	case []float32:
+		numDims, format, dimensionSize = len(values), C.DPI_VECTOR_FORMAT_FLOAT32, 4
+		if numDims > 0 {
+			valuesPtr = unsafe.Pointer(&values[0])
+		}
+	case []float64:
+		numDims, format, dimensionSize = len(values), C.DPI_VECTOR_FORMAT_FLOAT64, 8
+		if numDims > 0 {
+			valuesPtr = unsafe.Pointer(&values[0])
+		}
+	case []int8:
+		numDims, format, dimensionSize = len(values), C.DPI_VECTOR_FORMAT_INT8, 1
+		if numDims > 0 {
+			ptr := (*C.int8_t)(C.malloc(C.size_t(numDims)))
```

Comment:
What's the error here (why do we need malloc, why not just an unsafe.Pointer(&values[0])?